File: codility/lessons/sorting/Distinct.java
package codility.lessons.sorting;

import java.util.*;

/**
 * Level : painless
 */
class Distinct {
    public int solution(int[] A) {
        Set<Integer> set = new HashSet<>(A.length);
        for (int i=0 ; i<A.length ; set.add(A[i++]));
        return set.size();
    }
}

File: src/main/scala/codility/lessons/sorting/Distinct.scala
package codility.lessons.sorting

/**
 * Level : painless
 */
object DistinctScala {
    object Solution {
        // I know this is cheating, but why would I sort the array if I can do more efficient ? This problem
        // is pointless anyway
        def solution(arr: Array[Int]) = arr.toSet.size
    }
}